### DISCUSSION TITLE
Is it possible to generate an importable schema?

### DISCUSSION THREAD

**Discussion by GreenGremlin** [dotansimha/graphql-code-generator#5100] [Q&A] [answered]
I need an easy way to import the schema to pass it to @graphql-tools/mock's `addMocksToSchema`. I know I can use the schema-ast plugin to output the schema as a schema.graphql file, but importing the schema.graphql requires complicated loader setup for each environment. Is there a graphql-codegen plugin that will generate a TS / JS file that exports the graphQL schema as a string?

### Example `schema.ts` that I'm hoping to generate:

```ts
export const schema = `${SCHEMA_STRING}`;
```

Alternatively, the schema parsed as a `DocumentString`, as if it were imported using `graphql-tag/loader` would work. I tried the typed-document-node plugin, but it generates DocumentNodes for operations and not for the schema.

**Comment by dotansimha** [comment, marked as answer]
You can do something like that as a custom plugin:

```
const { printSchema } from 'graphql';

module.exports = {
  plugin: (schema) => {
    return {
      content: `export const schema = \`${printSchema(schema)}\`; `
    }
  }
}
```

**Reply by GreenGremlin** [reply]
Perfect! Thank you!
